test/utils/testconfig/testconfig_test.go

```diff
@@ -226,17 +233,29 @@ func mockPtpConfig(name, namespace string, role ptpv1.PtpRole, mode PTPMode) *pt
 		aStringPhc2sys := "-a -r -r"
 		aProfile.Phc2sysOpts = &aStringPhc2sys
 	} else {
-		aStringPtp4l := "-s"
-		aProfile.Ptp4lOpts = &aStringPtp4l
-		aStringPhc2sys := "-a -r"
-		aProfile.Phc2sysOpts = &aStringPhc2sys
-		aStringInterface := "eth0"
-		aProfile.Interface = &aStringInterface
+
 		switch mode {
+		case OrdinaryClock:
+			aStringPtp4l := "-s -2"
+			aProfile.Ptp4lOpts = &aStringPtp4l
+			aStringPhc2sys := "-a -r"
+			aProfile.Phc2sysOpts = &aStringPhc2sys
+			aStringInterface := "eth0"
+			aProfile.Interface = &aStringInterface
 		case BoundaryClock:
+			aStringPtp4l := "-2"
+			aProfile.Ptp4lOpts = &aStringPtp4l
+			aStringPhc2sys := "-a -r"
+			aProfile.Phc2sysOpts = &aStringPhc2sys
```

Comment:
There is new finding, we need to pass  -s slave_interface_name to phc2sysOpts in case of dual nic

Reply:
Is this change for "BC" and "dualnic BC main" nics? or just dualnic BC? For dual nic BC does it apply to both main/secondary nics, or just main?

Reply:
lets ignore this for now. Its fine

Reply:
right way to do it ` -m -O -37 -N 8 -R 16 -s <slave_interface> -n 24`, but that;s fine for now

Reply:
ok np